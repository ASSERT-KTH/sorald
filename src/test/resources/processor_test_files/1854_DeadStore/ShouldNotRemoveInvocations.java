/*
 * Test to verify that the processor does not remove method invocations by default,
 * as these may have side effects.
 */

public class ShouldNotRemoveInvocations {
    public void methodWithDeadStoreMethodCall() {
        int a = someMethod(); // Noncompliant
        a = otherMethod(); // Noncompliant
        a = 2;
        System.out.println(a);
    }
}