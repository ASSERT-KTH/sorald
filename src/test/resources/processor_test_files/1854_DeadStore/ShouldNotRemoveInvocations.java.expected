/*
 * Test to verify that the processor does not remove method invocations by default,
 * as these may have side effects.
 */

public class ShouldNotRemoveInvocations {
    public void methodWithDeadStoreMethodCall() {
        someMethod();
        otherMethod();
        int a = 2;
        System.out.println(a);
    }
}